activerecord/lib/active_record/relation/batches.rb

```diff
@@ -211,39 +211,41 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
       relation = apply_limits(relation, start, finish)
       batch_relation = relation
 
-      loop do
-        if load
-          records = batch_relation.records
-          ids = records.map(&:id)
-          yielded_relation = where(primary_key => ids)
-          yielded_relation.load_records(records)
-        else
-          ids = batch_relation.pluck(primary_key)
-          yielded_relation = where(primary_key => ids)
-        end
+      uncached do
+        loop do
+          if load
+            records = batch_relation.records
+            ids = records.map(&:id)
+            yielded_relation = where(primary_key => ids)
+            yielded_relation.load_records(records)
+          else
+            ids = batch_relation.pluck(primary_key)
+            yielded_relation = where(primary_key => ids)
+          end
 
-        break if ids.empty?
+          break if ids.empty?
 
-        primary_key_offset = ids.last
-        raise ArgumentError.new("Primary key not included in the custom select clause") unless primary_key_offset
+          primary_key_offset = ids.last
+          raise ArgumentError.new("Primary key not included in the custom select clause") unless primary_key_offset
 
-        yield yielded_relation
+          yield yielded_relation
```

Comment:
I think this might surprise people, by spreading the uncaching too far.

It rather feels like we might need to add a "don't cache this" flag to the relation itself.